src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/service/FlowCacheService.java

```diff
@@ -49,9 +52,8 @@ public FlowCacheService(PersistenceManager persistenceManager, Clock clock,
         this.clock = clock;
         this.flowRttStatsExpirationTime = flowRttStatsExpirationTime;
         this.carrier = carrier;
-
-        FlowRepository flowRepository = persistenceManager.getRepositoryFactory().createFlowRepository();
-        initCache(flowRepository);
+        flowRepository = persistenceManager.getRepositoryFactory().createFlowRepository();
+        activate();
```

Comment:
Yes, previously we initialized cache in the constructor, but it's not right approach.
It must be activated only by calling `public void activate()`

By default each topology is inactive. We have special zookeeper states to activate/deactivate it https://github.com/telstra/open-kilda/blob/ac2db6a2dc88a7cde627b6ae5c2c3621165431a2/docker/zookeeper/bin/create-zknode-and-start.sh#L98

With the help of these state we can turn off and on topologies. If zookeeper will has signal "SHUTDOWN" this cache must be empty, But now - it will be initialized during the start.

Reply:
done